docs/development/logging.md

```diff
@@ -0,0 +1,165 @@
+# Logging in Gardener Components
```

Comment:
Can you add this document to the index, please?

Reply:
Oh for sure... 😄